source/pdal/pdalc_pipeline.cpp

```diff
@@ -26,70 +26,73 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  *****************************************************************************/
-
+#define _CRT_SECURE_NO_WARNINGS
 #include "pdalc_pipeline.h"
 
 #include <string>
 
-#include <pdal/PipelineExecutor.hpp>
-
 #include "pdalc_pointviewiterator.h"
 
-// TODO Address cause of std::min problems. See https://github.com/PDAL/CAPI/issues/4
+#include <pdal/util/Utils.hpp>
+#include <pdal/PipelineWriter.hpp>
+#include <pdal/Stage.hpp>
+#include <pdal/pdal_types.hpp>
+
 #undef min
 
 namespace pdal
 {
 namespace capi
 {
+
 extern "C"
 {
-    PDALPipelinePtr PDALCreatePipeline(const char* json)
+    struct Pipeline
     {
-        PDALPipelinePtr pipeline = nullptr;
+    public:
+
+        PipelineManagerPtr manager = std::make_unique<PipelineManager>();
 
+        bool m_executed = false;
+        std::stringstream logStream;
+    };
+
+    PDALPipelinePtr PDALCreatePipeline(const char* json)
+    {
+        Pipeline* pipeline = new Pipeline();
```

Comment:
why use unique_ptr for the Manager but not for this one?

Reply:
The ABI (which has not been changed) depends on the Pipeline instance persisting between stateless C calls until manually disposed by the consuming app. This is, as I understand it, WAI.

Reply:
WAI, meaning... ?

Reply:
"Working As Intended" :) I think that is me failing automatically into a particular corps short hand :)

Reply:
I thought some more about your comment and realised what you meant and created a hopefully better and safer version